providers/base/bin/suspend_stats.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+# This file is part of Checkbox.
+#
+# Copyright 2025 Canonical Ltd.
+# Written by:
+#   Hanhsuan Lee <hanhsuan.lee@canonical.com>
+#
+# Checkbox is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# Checkbox is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Checkbox.  If not, see <http://www.gnu.org/licenses/>.
+import argparse
+import sys
+import os
+
+
+class SuspendStats:
+    """
+    This class is used to parse the information under
+    /sys/power/suspend_stats/
+
+    """
+
+    contents = {}
+
+    def __init__(self):
+        suspend_stat_path = "/sys/power/suspend_stats/"
+        self.contents = self.collect_content_under_directory(suspend_stat_path)
+
+    def collect_content_under_directory(self, search_directory: str) -> dict:
+        """
+        Collect all content under specific directory by filename
+
+        :param search_directory: The directory to search through.
+
+        :returns: collected content by filename
+        """
+        content = {}
+        for root, dirs, files in os.walk(search_directory):
+            for file_name in files:
+                file_path = os.path.join(root, file_name)
+                with open(
+                    file_path, "r", encoding="utf-8", errors="ignore"
+                ) as file:
+                    content[file_name] = file.read().splitlines()[0]
+        return content
```

Comment:
Slightly easier to read imo
```suggestion
        search_directory = Path(search_directory)
        for p in filter(lambda x: x.is_file(), search_directory.iterdir()):
            content[p.name], *_ = p.read_text().splitlines(1)
        return content
```